### ISSUE TITLE
Volume label doesn't support any special characters such as underscore

### ISSUE THREAD

**Issue by distantvale** [dosbox-staging/dosbox-staging#4118]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Windows 11

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

When mounting and using the -label option with a specific volume label, special characters are removed before the label is set.
This includes characters such as underscore which should be supported.
Also see this -
https://jeffpar.github.io/kbarchive/kb/026/Q26295/

### Steps to reproduce the behaviour.

Explain how to reproduce
1. mount any drive with "-label A_B" option
2. navigate to the drive and run the "dir" command
3. see that the volume label is actually set as "AB" (underscore omitted)

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by johnnovak** [comment]
All yours @NicknineTheEagle

**Comment by NicknineTheEagle** [comment]
https://github.com/dosbox-staging/dosbox-staging/blob/e5c8d3a24e2834fb2898dde1517e982d025c7934/src/dos/drives.cpp#L28-L45

Well, there's the problem, To_Label function used by DIR command calls strip_punctuation which removes all characters that std::ispunct returns true for. This issue is actually acknowledged in the comment and it also points out that this issue is kind of a mess.

**Comment by johnnovak** [comment]
> https://github.com/dosbox-staging/dosbox-staging/blob/e5c8d3a24e2834fb2898dde1517e982d025c7934/src/dos/drives.cpp#L28-L45
> 
> Well, there's the problem, To_Label function used by DIR command calls strip_punctuation which removes all characters that std::ispunct returns true for. This issue is actually acknowledged in the comment and it also points out that label formatting is kind of a mess.

Cool, please PR the change then (you said you wanted to work on this).

**Comment by johnnovak** [comment]
@NicknineTheEagle @distantvale Can you guys extend the top description with an actual concrete where this caused a problem? You couldn't mount a game installer CD or something? Which game?

**Comment by distantvale** [comment]
@johnnovak I don't recall a specific test for CD but I did see old original CDs with underscore in their label and I just added a link to one of them in the top description.

**Comment by Dustman0192** [comment]
Hello! I'm just getting into helping out with Dosbox and I believe I have a fix for this along with some new unit tests for it. Should I just check in my branch and raise a PR? I'm not yet familiar with the process behind how you guys do PRs (though I am quite familiar with how it's done in the enterprise world :) )

**Comment by johnnovak** [comment]
> Hello! I'm just getting into helping out with Dosbox and I believe I have a fix for this along with some new unit tests for it. Should I just check in my branch and raise a PR? I'm not yet familiar with the process behind how you guys do PRs (though I am quite familiar with how it's done in the enterprise world :) )

@phoenixno624 Yes, we welcome code contributions and are very interested in your solution.

As you said, creating a feature branch in your fork and raising a PR is the way. This document gives a good overview of our development standards for new code contributors:

https://github.com/dosbox-staging/dosbox-staging/blob/main/docs/CONTRIBUTING.md

Looking forward to your PR! 🙂

**Comment by Dustman0192** [comment]
@johnnovak Thanks! This is actually the first C++ open source project I've ever contributed to but I'm totally looking forward to helping out! I've raised my PR and filled out what I could. Totally let me know if I missed anything or if anything could be improved!

**Comment by Torinde** [comment]
Related:
- https://github.com/schellingb/dosbox-pure/issues/709
- https://github.com/schellingb/dosbox-pure/commit/38c84798aac5239a06d75e85469225a2ad9c4707
- https://github.com/dosbox-staging/dosbox-staging/discussions/4734

**Comment by johnnovak** [comment]
@dividebysandwich Might be relevant to your interests.
